iceoryx_posh/source/runtime/service_discovery.cpp

```diff
@@ -44,18 +44,25 @@ ServiceContainer ServiceDiscovery::findService(const cxx::optional<capro::IdStri
     return searchResult;
 }
 
-void ServiceDiscovery::findService(const cxx::optional<capro::IdString_t>& service,
-                                   const cxx::optional<capro::IdString_t>& instance,
-                                   const cxx::optional<capro::IdString_t>& event,
-                                   const cxx::function_ref<void(const ServiceContainer&)>& callable) noexcept
+void ServiceDiscovery::findService(
+    const cxx::optional<capro::IdString_t>& service,
+    const cxx::optional<capro::IdString_t>& instance,
+    const cxx::optional<capro::IdString_t>& event,
+    const cxx::function_ref<void(const capro::ServiceDescription&)>& callableForEach) noexcept
 {
-    if (!callable)
+    if (!callableForEach)
```

Comment:
This check is so painful to watch... we need to get rid of it with a better design (e.g. by disallowing null functions) or by just an assert that states the contract. Aware users will never pass a null function ... and then it causes only costs, which depending on the CPU pipeline may be insignificant if we are waiting for something else but it is an extra instruction cycle that can be used for something unrelated ...

Anyway, nothing to be done now, resolve me.